plugins/embark-assistant/matcher.cpp

```diff
@@ -1515,16 +1518,20 @@ namespace embark_assist {
                     break;
 
                 case embark_assist::defs::aquifer_ranges::None_Plus_Light:
-                    if (!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)) ||
-                        !(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light))) {
+                    if ((!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)) &&
+                        (!(static_cast<int8_t>(tile->neighboring_aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)))) ||
+                        (!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light)) &&
+                            (!(static_cast<int8_t>(tile->neighboring_aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light))))) {
                         if (trace) out.print("matcher::world_tile_match: Aquifer None_Plus_Light (%i, %i)\n", x, y);
                         return false;
                     }
                     break;
 
                 case embark_assist::defs::aquifer_ranges::None_Plus_At_Least_Light:
-                    if (!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)) ||
-                        !(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light_Heavy))) {
+                    if ((!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)) &&
+                        (!(static_cast<int8_t>(tile->neighboring_aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::None)))) ||
+                        (!(static_cast<int8_t>(tile->aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light_Heavy)) &&
+                            (!(static_cast<int8_t>(tile->neighboring_aquifer) & static_cast<int8_t>(embark_assist::defs::aquifer_sizes::Light_Heavy))))) {
```

Comment:
I might suggest wrapping some of these checks in a function, if there's a way to do that that makes sense (it could help reduce the `)))))` at the end, and possibly abstract away some of the negations). Up to you.

I'm also wondering if `aquifer_sizes` would make more sense as a bitfield instead of an enum, but there's no need to change it here (just pointing out that checking for specific bits in a bitfield could avoid the need for casting to `int8_t`).

Reply:
The reason for the use of an enum is primarily that this is what makes sense in the UI, so that is what was used throughout. However, it would probably have been better to translate it into a bit field once it gets out of the UI.
As usual, it started with only a few cases where it mattered, and then I had to add a whole bunch more...

Reply:
As you can see, I reworked the aquifer bit handling from an enum to named bit constants, but didn't go all the way to a bit field. At the same time, I realized I could merge the native tile aquifer info with the neighboring tiles one so there would only be a single check against the criterion at hand. I think it was only one section of the code where the checks
became slightly longer, where several became considerably shorter.

Reply:
Yeah, I figured it was UI-related. I like what you came up with here!

Reply:
Thanks.

I assume it is as you assumed, i.e. that I may have pulled the master rather than develop. It sure would lessen confusion if the same convention was used throughout, though.

And thanks for sorting things out.

Reply:
For reference, I think https://docs.dfhack.org/en/stable/docs/Contributing.html#pull-request-guidelines is up-to-date (and would change if the default branch names change)

Reply:
If you read the documentation every time, and aren't cocky enough to think you remember from the last time, a number of months earlier, yes...